### PR TITLE
feat(FieldSelector): Add optional histogram, sorted setting

### PR DESCRIPTION
**Pull request by aronhelser**
@jourdain @vibraphone @scottwittenburg Seb, you may be the only one not on vacation, but I know you're busy! If you get a chance to look...

Add a small histogram and min/max entry for each field.
Only appears if model.fieldShowHistogram is true.

Add a sorted setting, so FieldProvider sorts names before
returning them. Histogram example uses sorting.

Add window resize handling. FieldSelector min/max and histogram
will hide/show dynamically as window size changes.

Add field count to the header label.

Add uncertainty to HistogramSelector dividers - not used yet.

### CODE REVIEW

File: src/InfoViz/Core/FieldProvider/index.js [hunk 1]
```diff
@@ -13,9 +13,19 @@ function fieldProvider(publicAPI, model) {
     model.fields = {};
   }
 
-  publicAPI.getFieldNames = () => Object.keys(model.fields);
+  publicAPI.setFieldsSorted = (doSort) => (model.doSort = doSort);
```
Comment (jourdain) [inline src/InfoViz/Core/FieldProvider/index.js]: Use the setGet helper to handle that function instead.

File: src/InfoViz/Native/FieldSelector/index.js [hunk 2]
```diff
@@ -13,19 +14,48 @@ import template from './template.html';
...
+      model.fieldShowHistogram = model.fieldShowHistogram && (model.provider.isA('Histogram1DProvider'));
+      // append headers for histogram columns
+      if (model.fieldShowHistogram) {
+        const header = d3.select(model.container).select('thead').select('tr');
+        header.append('th').text('Min').classed(style.jsHistMin, true);
+        header.append('th').text('Histogram').classed(style.jsSparkline, true);
+        header.append('th').text('Max').classed(style.jsHistMax, true);
+      }
+      // Listen to window resize
+      model.sizeSubscription = sizeHelper.onSizeChange(publicAPI.resize);
```
Comment (jourdain) [inline src/InfoViz/Native/FieldSelector/index.js]: Not sure we want the component to do that. It should probably managed at the parent level (application/example).

**Comment by jourdain** [comment]
I've added some comments in the file changed section.

File: src/InfoViz/Native/FieldSelector/index.js (hunk 2 above)
Comment (aronhelser) [inline src/InfoViz/Native/FieldSelector/index.js]: The parent can manage it, by overriding model.fieldShowHistogram. Maybe it should be a getter/setter, too?

Comment (jourdain) [inline src/InfoViz/Native/FieldSelector/index.js]: I was talking about the resize / sizeHelper part

Comment (aronhelser) [inline src/InfoViz/Native/FieldSelector/index.js]: OK, that works fine too :) I've pushed the fix

**Comment by aronhelser** [comment]
@jourdain @vibraphone Is this good to merge?

**Pull request merged**